### PR TITLE
Add support for nist P384 and P521 curves, as well as the brainpool 256, 384 and 512 bit curves

### PR DESCRIPTION
**Pull request by sosthene-nitrokey**
Pending: https://github.com/trussed-dev/trussed/pull/163

### CODE REVIEW

**Comment by jans23** [comment]
Does this support SE050?

**Comment by sosthene-nitrokey** [comment]
Yes. It fully works with the software implementation, but it also contains a workaround for an issue that comes with the SE050 implementation and private key import.

The other relevant PRs are: 
- https://github.com/Nitrokey/nitrokey-3-firmware/pull/484
- https://github.com/Nitrokey/trussed-se050-backend/pull/33
- https://github.com/trussed-dev/trussed/pull/163

Due to binary size constraints, we are going to only support these curves when the SE050 anyways on the NK3 firmware (at least until we make significant progress on that front).

File: src/card.rs [hunk 1]
```diff
@@ -159,42 +159,71 @@ impl<T: Client, const C: usize, const R: usize> apdu_dispatch::App<C, R> for Car
...
+        /// RSA 3072
+        const RSA_3072 = 1 << 4;
+        /// RSA 4096
+        const RSA_4096 = 1 << 5;
+        /// X25519
+        const X_25519 = 1 << 6;
+        /// EdDsa25519
+        const ED_25519 = 1 << 7;
+        /// BRAINPOOL_P256R1 Brainpool curve
+        const BRAINPOOL_P256R1 = 1<<8;
```
Comment (daringer) [inline src/card.rs:185]: tiny nit - `cargo fmt` doesn't catch this ?

```suggestion
        const BRAINPOOL_P256R1 = 1 << 8;
```

Comment (sosthene-nitrokey) [reply src/card.rs]: I suppose no because of the `bitflags` macro.

**Pull request merged**